common/src/main/java/com/wynntils/core/config/Configurable.java

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.config;
+
+/** Used to indicate that a class contains config options */
+public interface Configurable {
+    default void onUpdate() {}
```

Comment:
Ok, now I understand how this works. Sorry if my comments appear out of order and a bit confused. :-)

I'm not 100% sure the best way is to have a default implementation. I'm thinking there is at least a strong argument for forcing all features that have configs to implement this method, to force them to think about what would happen if the config updates. 

But then again maybe that's too much of a burden; perhaps most features will just read the value of the config object when it is needed.

I'm on the fence here...

Reply:
Oh, and also, can you name the method a bit less generic? :-) Like `onConfigurationUpdate` or such.

Reply:
> Ok, now I understand how this works. Sorry if my comments appear out of order and a bit confused. :-)
> 
> 
> 
> I'm not 100% sure the best way is to have a default implementation. I'm thinking there is at least a strong argument for forcing all features that have configs to implement this method, to force them to think about what would happen if the config updates. 
> 
> 
> 
> But then again maybe that's too much of a burden; perhaps most features will just read the value of the config object when it is needed.
> 
> 
> 
> I'm on the fence here...

I think most features wouldn't use the onUpdate method, iirc in legacy it is mostly used to force config save updates, and reload a few guis (which won't be an issue with proper design).

Reply:
Ok, let's keep this with the default empty implementation then.